frontend/src/api/TokenRefreshInstance.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { FIVETHOUSAND, THOUSAND } from '../constants/valuableNumbers'
+import {HttpClient} from './HttpClient'
+
+interface IToken {
+    access:string
+token: {
+    access: string
+    raw: string
+    payload: {
+        exp: number
+        id: number
+    }
+}
+split: any
+}
+
+interface ITokenSend {
+    refresh: string | null
+}
+
+interface ITokenRefresh {
+    access: string
+    refresh: string
+}
+
+export interface ISendTokenRefresh {
+    postRefresh: (payload: ITokenSend) => Promise<ITokenRefresh>
+}
+
+export function jwtDecode(t: IToken) {
+    const token = {
+        raw: {},
+        payload: {},
+    }
+    token.raw = t
+    token.payload = JSON.parse(window.atob(t.split('.')[1]))
+    return (token)
+  }
+
+export class TokenRefresh extends HttpClient implements ISendTokenRefresh {    
+    async postRefresh(payload: ITokenSend): Promise<ITokenRefresh> {
+        return await this.post('token/refresh', true, payload) as ITokenRefresh
```

Comment:
I know it's little bit out of scope, but let's update `HttpClient` so 2nd argument won't be just a boolean, but an object. Using just a boolean looks bit confusing because does not give any context. Changing it to something like `type HttpRequestOptions = { useCredentials: boolean }` will give us more context and power to enhance `HttpClient` functional in future.

Also, as I remember, `get` and `post` are generics, so you can call it like:
```
await this.post<ITokenSend, ITokenResresh>this.post(...)
```
So you don't need `as`

Reply:
Not understand about HttpRequestOptions object in headers. Need it be a separate object inside HttpClient method?